source/simplex/fe_lib.cc

```diff
@@ -239,6 +241,65 @@ namespace Simplex
   }
 
 
+
+  template <int dim, int spacedim>
+  FiniteElementDomination::Domination
+  FE_P<dim, spacedim>::compare_for_domination(
+    const FiniteElement<dim, spacedim> &fe_other,
+    const unsigned int                  codim) const
+  {
+    (void)fe_other;
+    (void)codim;
+
+    Assert((dynamic_cast<const FE_Q<dim, spacedim> *>(&fe_other)),
+           ExcNotImplemented());
+    AssertDimension(dim, 2);
```

Comment:
I think it makes sense to indicate what the potential problem is, namely the impossibility to set up an interpolation between a 3D FE_Q and a 3D FE_P due to the different types of faces. How about this message:
```suggestion
    Assert(dim == 2, ExcNotImplemented("hp comparision not possible for dim=" + std::to_string(dim)));
```

Reply:
Let's keep this assert. It will be remove quite soon anyway (in the next 1-2 weeks when wedges and pyramids are coming).